fineract-core/src/main/java/org/apache/fineract/infrastructure/core/service/database/DatabaseSpecificSQLGenerator.java

```diff
@@ -36,14 +42,26 @@ public DatabaseSpecificSQLGenerator(DatabaseTypeResolver databaseTypeResolver) {
     }
 
     public String escape(String arg) {
-        if (databaseTypeResolver.isMySQL()) {
+        return escape(databaseTypeResolver.databaseType(), arg);
+    }
+
+    public static String escape(DatabaseType dialect, String arg) {
+        if (dialect.isMySql()) {
             return format("`%s`", arg);
-        } else if (databaseTypeResolver.isPostgreSQL()) {
+        } else if (dialect.isPostgres()) {
             return format("\"%s\"", arg);
         }
         return arg;
     }
 
+    public String formatValue(JdbcJavaType columnType, String value) {
+        return formatValue(databaseTypeResolver.databaseType(), columnType, value);
+    }
+
+    public static String formatValue(DatabaseType dialect, JdbcJavaType columnType, String value) {
```

Comment:
How is this specific again to a dialect?

Reply:
I dont see that, the parameter is not even used.

Reply:
Same arguments here, see my previous comments "escape" function.